extensions/notebook/src/dialog/managePackages/addNewPackageTab.ts

```diff
@@ -0,0 +1,264 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as nls from 'vscode-nls';
+import * as azdata from 'azdata';
+import * as request from 'request';
+
+import JupyterServerInstallation from '../../jupyter/jupyterServerInstallation';
+import * as utils from '../../common/utils';
+import { ManagePackagesDialog } from './managePackagesDialog';
+
+const localize = nls.loadMessageBundle();
+
+export class AddNewPackageTab {
+	private addNewPkgTab: azdata.window.DialogTab;
+
+	private newPackagesSearchBar: azdata.InputBoxComponent;
+	private packagesSearchButton: azdata.ButtonComponent;
+	private newPackagesName: azdata.TextComponent;
+	private newPackagesNameLoader: azdata.LoadingComponent;
+	private newPackagesVersions: azdata.DropDownComponent;
+	private newPackagesVersionsLoader: azdata.LoadingComponent;
+	private newPackagesSummary: azdata.TextComponent;
+	private newPackagesSummaryLoader: azdata.LoadingComponent;
+	private packageInstallButton: azdata.ButtonComponent;
+
+	private readonly InstallButtonText = localize('managePackages.installButtonText', "Install");
```

Comment:
super minor: I think elsewhere we stick these as const values outside the class, or else just inline them. Up to you, would be nice to be consistent across our stack.